[CI] RejectedExecutionTests testHistoryOnRejection fails

Log: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.4+multijob-unix-compatibility/os=fedora-29&&immutable/50/console

Build scan: https://gradle-enterprise.elastic.co/s/lkpncm2tvpqii

Failure:

```
com.carrotsearch.randomizedtesting.UncaughtExceptionError: Captured an uncaught exception in thread: Thread[id=95, name=elasticsearch[node_s2][generic][T#2], state=RUNNABLE, group=TGRP-RejectedExecutionTests]Close stacktrace
at __randomizedtesting.SeedInfo.seed([2C8460A656D442E1:538BCEA5D73CC9A2]:0)
Caused by: org.elasticsearch.common.util.concurrent.EsRejectedExecutionException: rejected execution of processing of [79][indices:data/write/bulk[s][p]]: request: BulkShardRequest [[.triggered_watches][0]] containing [delete {[.triggered_watches][_doc][pXWYF_4d32327a-3e4a-404d-b5b8-d5927e7d5ed5-2019-09-24T09:58:00.03Z]}], target allocation id: 3HgeE2uCRmK-F1nbvAJGPA, primary term: 1 on EsThreadPoolExecutor[name = node_s1/write, queue capacity = 1, org.elasticsearch.common.util.concurrent.EsThreadPoolExecutor@6460c6a2[Running, pool size = 1, active threads = 1, queued tasks = 0, completed tasks = 4]]
```

Reproduce with:

```
./gradlew ':x-pack:plugin:watcher:test' --tests "org.elasticsearch.xpack.watcher.test.integration.RejectedExecutionTests.testHistoryOnRejection" -Dtests.seed=2C8460A656D442E1 -Dtests.security.manager=true -Dtests.locale=da-DK -Dtests.timezone=Europe/Gibraltar -Dcompiler.java=12 -Druntime.java=8
```

Cannot reproduce locally.

--- a/x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/execution/ExecutionService.java
+++ b/x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/execution/ExecutionService.java
@@ -435,11 +435,11 @@ public class ExecutionService {
                 }
                 try {
                     deleteTrigger(triggeredWatch.id());
-                } catch (Exception exc2) {
+                } catch (Exception exc) {
                     logger.error((Supplier<?>) () ->
                         new ParameterizedMessage(
                             "Error deleting entry from .triggered_watches for watch [{}] after thread pool rejection",
-                            triggeredWatch.id()), exc2);
+                            triggeredWatch.id()), exc);
                 }
             }));
         }

--- a/x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/execution/ExecutionService.java
+++ b/x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/execution/ExecutionService.java
@@ -433,7 +433,14 @@ public class ExecutionService {
                             "Error storing watch history record for watch [{}] after thread pool rejection",
                             triggeredWatch.id()), exc);
                 }
-                deleteTrigger(triggeredWatch.id());
+                try {
+                    deleteTrigger(triggeredWatch.id());
+                } catch (Exception exc2) {
+                    logger.error((Supplier<?>) () ->
+                        new ParameterizedMessage(
+                            "Error deleting entry from .triggered_watches for watch [{}] after thread pool rejection",
+                            triggeredWatch.id()), exc2);
+                }
             }));
         }
     }

